File: Pset6/readability py/readability.py
from cs50 import get_string 


def main():
    text = get_string("Text: ")
    letters, words, sentences = counting(text)
    degree = calculate_Degree(letters, words, sentences)
    if degree >= 1 and degree <= 16:
        print(f"Grade {degree}")
    elif degree < 1:
        print("Before Grade 1")
    else:
        print("Grade 16+")

    
def counting(s):
    words = 1
    letters = sentences = 0
    for c in s:
        if c.isalpha():
            letters += 1
        elif c in [" "]:
            words += 1
        elif c in ["!", ".", "?"]:
            sentences += 1
    #print(letters, words, sentences)
    return letters, words, sentences

 
def calculate_Degree(l, w, s):
    l = 100 * l / w
    s = 100 * s / w
    #print(l, w, s)
    return round(0.0588 * l - 0.296 * s - 15.8)
    
    
if __name__ == "__main__":
    main()

File: Pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int numberofletters(string text);
int numberofwords(string text);
int numberofsentences(string text);
int Coleman_Liau(float l, float w, float s);

int main(void)
{
    int letterNumber, wordNumber, sentenceNumber, grade;
    string text = get_string("Text: ");
    letterNumber = numberofletters(text);
    wordNumber = numberofwords(text);
    sentenceNumber = numberofsentences(text);
    grade = Coleman_Liau((float)letterNumber, (float)wordNumber, (float)sentenceNumber);
    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade <= 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

int numberofletters(string text)
{
    int n = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            n++;
        }
    }
    return n;
}

int numberofwords(string text)
{
    int n = 1;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
        {
            n++;
        }
    }
    return n;
}

int numberofsentences(string text)
{
    int n = 0;
    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == '!' || text[i] == '?' || text[i] == '.')
        {
            n++;
        }
    }
    return n;
}

int Coleman_Liau(float l, float w, float s)
{
    l = 100 * l / w;
    s = 100 * s / w;
    return round(0.0588 * l - 0.296 * s - 15.8);
}